### DISCUSSION TITLE
Ion Storms and Space-Dust clouds

### DISCUSSION THREAD

**Discussion by eldinor** [jelster/space-truckers#24] [Playground Snippets]
Dust cloud that follows a path, like a nebula. Or asteroid belt (also possible to use SPS)

https://playground.babylonjs.com/#R1JWLA#36

![image](https://user-images.githubusercontent.com/25833327/106367681-67379d00-6355-11eb-9c73-1e000b657343.png)

There are some tricks in this example, see discussion
https://forum.babylonjs.com/t/noise-particle-cloud-following-and-orbiting-a-path/17971

**Comment by jelster** [comment]
I like this for showing solar flares and other activity! We could model the motion akin to charged particles moving in a magnetic field with some pretty simple parametric functions I think

**Reply by jelster** [reply]
For the asteroids in the Phase 1 game state, I built [this snippet](https://playground.babylonjs.com/#9EI3H4%231) that uses the SPS and some random maths to generate the rocks. The generator could use some tweaking to reduce the "pancake" asteroids that sometimes result.

**Comment by jelster** [comment]
Solar wind particles twirl and spin around and through the road "track" which is actually a tube. The tube itself is twisted and warped where it intersects with the wind, reflecting the intense bath of solar wind particles tangling themselves with it.
